actions/form/class.Instance.php

```diff
@@ -227,4 +230,42 @@ private function isEmptyLabel($element): bool
         return $element instanceof tao_helpers_form_elements_Label
             && empty($element->getRawValue());
     }
+
+    /**
+     * Check if the Solar design is enabled and the postfix has not yet been added
+     *
+     */
+    private function isContainPostfix(string $language): bool
+    {
+        $pattern = '/' . self::LANG_POSTFIX . '$/';
+
+
+        return !Layout::isSolarDesignEnabled() || preg_match($pattern, $language, $matches);
+    }
+
+    /**
+     * Concatenate postfix for Solar design translations
+     *
+     */
+    private function addPostfix(string $language): string
+    {
+        return $language . self::LANG_POSTFIX;
+    }
+
+    /**
+     * Check and Add postfix for Solar design translations
+     *
+     */
+    private function checkPostfix(string $language): string
+    {
+        if (!$this->isContainPostfix($language)) {
+            $localesDir = 'views/locales';
+            $dir = dirname(__FILE__) . '/../../' . $localesDir . '/' . $this->addPostfix($language);
+            if (is_dir($dir)) {
+                $language = $this->addPostfix($language);
+            }
+        }
+
+        return $language;
```

Comment:
We can simplify it a little, wdyt?
```suggestion
        if ($this->isContainPostfix($language)) {
            return $language;
        }
        $localesDir = 'views/locales';
        $dir = sprintf('%s/../../%s/%s', dirname(__FILE__), $localesDir, $this->addPostfix($language));
        if (is_dir($dir)) {
                return $this->addPostfix($language);
        }
        return $language;
```